pyrokinetics/equilibrium/equilibrium.py

```diff
@@ -321,11 +321,17 @@ def __init__(
         R_major = np.asfarray(R_major) * eq_units["len"]
         r_minor = np.asfarray(r_minor) * eq_units["len"]
         Z_mid = np.asfarray(Z_mid) * eq_units["len"]
+
+        B_0 = B_0 * cocos_factors["BT"] * eq_units["B"]
+        I_p = I_p * cocos_factors["IP"] * eq_units["I"]
```

Comment:
An issue with moving these here is that it'll fail for equilibria where we can't define `B_0` and `I_p`, as they're `None` by default. We could upgrade these to non-optional positional arguments, or we could have something like:

```python
Ip_sign = 1 if I_p is None else int(np.sign(I_p))
```
The bit at the bottom of the function where `B_0` and `I_p` are added to `attrs` would then be the same as it was.